dpnp/dpnp_iface_statistics.py

```diff
@@ -747,26 +803,95 @@ def var(x1, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
     >>> import dpnp as np
     >>> a = np.array([[1, 2], [3, 4]])
     >>> np.var(a)
-    1.25
+    array(1.25)
+    >>> np.var(a, axis=0)
+    array([1.,  1.])
+    >>> np.var(a, axis=1)
+    array([0.25,  0.25])
 
     """
 
-    x1_desc = dpnp.get_dpnp_descriptor(x1, copy_when_nondefault_queue=False)
-    if x1_desc:
-        if x1_desc.size == 0:
-            pass
-        elif axis is not None:
-            pass
-        elif dtype is not None:
-            pass
-        elif out is not None:
-            pass
-        elif keepdims:
-            pass
+    if where is not True:
+        raise NotImplementedError(
+            "where keyword argument is only supported with its default value."
+        )
+    else:
+        if issubclass(a.dtype.type, dpnp.complexfloating):
+            # Note that if dtype is not of inexact type then arrmean will not be either.
+            arrmean = dpnp.mean(
+                a, axis=axis, dtype=dtype, keepdims=True, where=where
+            )
+            x = dpnp.subtract(a, arrmean, where=where)
+            x = dpnp.multiply(x, x.conj(), where=where).real
+            res = dpnp.sum(
+                x, axis=axis, dtype=dtype, keepdims=keepdims, where=where
+            )
+            if dtype is None and a.real.dtype != res.dtype:
+                res = res.astype(a.real.dtype, casting="same_kind")
+
+            cnt = _count_reduce_items(a, axis, where)
+            dof = dpnp.max(
+                dpnp.array(
+                    [cnt - ddof, 0],
+                    dtype=res.dtype,
+                    sycl_queue=res.sycl_queue,
+                    device=res.device,
+                )
+            )
```

Comment:
`dot` looks as a scalars, do we need to create dpnp array here to calculate max of two scalars?
should we use numpy.max or built-in `max` instead?